packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -7924,6 +7985,16 @@ class AdapterClass extends EventEmitter {
          */
         this.unsubscribeStatesAsync = tools.promisify(this.unsubscribeStates, this);
 
+        /**
+         * Promise-version of Adapter.subscribeForeignFiles
+         */
+        this.subscribeForeignFilesAsync = tools.promisify(this.subscribeForeignFiles, this);
```

Comment:
Do we want to make new methods promise only? I would support this.

Reply:
But especially for subscribes nooen really use callbacks or await or? ;-))

Reply:
Not really necessary except you want to check if the subscribe succeeded, but then still no need for two methods.

Reply:
Done